userspace/libsinsp/filter_compare.cpp

```diff
@@ -583,9 +583,36 @@ bool flt_compare_ipv6net(cmpop op, const ipv6addr* operand1, const ipv6net* oper
 // flt_cast takes a pointer to memory, dereferences it as fromT type and casts it
 // to a compatible toT type
 template<class fromT, class toT>
-static inline toT flt_cast(const void* ptr) {
-	fromT val;
-	memcpy(&val, ptr, sizeof(fromT));
+static inline toT flt_cast(const void* ptr, uint32_t len) {
+	fromT val{};
+	/*
+	 * In big endian systems, we need to
+	 * make sure that we copy the right bytes
+	 * when len > sizeof(fromT).
+	 * This is an edge case that should only happen
+	 * for `evt.rawarg.*` fields.
+	 */
+	uint8_t shift = 0;
+#ifdef __s390x__
+	// NOTE: c++20 has `constexpr (std::endian::native == std::endian::big)`
+	// that would be much better.
+	// To avoid perf hit, only compile this on s390x (our only big-endian supported arch).
+	if(len > sizeof(fromT)) {
+		shift = len - sizeof(fromT);
+	}
+#endif
+
+	/*
+	 * Another fix for `evt.rawarg.*` fields:
+	 * it can happen that we evaluated eg: `evt.rawarg.flags` to be uin16_t at filter compile time,
+	 * but then when we extract from event, we expect an uint32_t.
+	 * Without this check, we would try to copy 4B of data while our ptr only holds 2B of data.
+	 */
+	size_t size = sizeof(fromT);
+	if(len > 0 && len < size) {
```

Comment:
It happens that we reach this with `len == 0`; that is the default value given by 
```
bool compare_rhs(cmpop op, ppm_param_type type, const void* operand1, uint32_t op1_len = 0);
```
in sinsp_filtercheck.h.
When len is 0, we expect the size to be the requested one. Otherwise, i could also update the usages of `compare_rhs()` requiring a proper size to be passed for int types, and put some ASSERTs in the switch case.